src/v/v8_engine/code_database.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "hashing/xx.h"
+#include "model/metadata.h"
+
+#include <absl/container/node_hash_map.h>
+
+namespace v8_engine {
+
+class code_database {
+public:
+    void insert_or_assign(size_t id, iobuf code) {
+        _code_db.insert_or_assign(id, std::move(code));
+    }
+
+    void erase(size_t id) { _code_db.erase(id); }
+
+    std::optional<iobuf> get_code(std::string_view name) const {
+        // rpk use xxhash_64 for create script_is from script name
```

Comment:
the script id is already a hash of the name

Reply:
But data_policy contains script_name, not script_id( So we need to calculate it